test/unit/active_model/array_serializer/serialization_test.rb

```diff
@@ -49,6 +49,7 @@ def test_serializer_for_array_returns_appropriate_type
         def object.serializer_class; CustomSerializer; end
 
         assert_equal CustomSerializer, Serializer.serializer_for(object)
+        assert_equal CustomSerializer, Serializer.serializer_for('Custom')
```

Comment:
this is the failing test, right?

is accepting a String a new behavior?

Reply:
This test is fine. I added an extra test since i want to allow string instead of a whole class object. This somewhat relaxes the input for serializer_for. Not sure if its a good idea, but its one simple fix i can think off.

Upon upgrading to ruby 2.5 >, there are 17 tests that fails due to the const_get changes. I used thou failed test as a benchmark to validate if the fix work. Is that ok? Is there a need to include new test?